### PR TITLE
New Record Medium Track: 14.34min (-470 steps, -90s) MUDD Connections & MUDD Style Gate 

### PR DESCRIPTION
**Pull request by Lisennlp**
Add MUDD (Multiway Dynamic Dense Connections, [Xiao et al. 2025](https://arxiv.org/abs/2502.12170)) to the **last two transformer blocks** of the Medium speedrun model. The idea matches the Small-track PR #259 : trim routing cost, reuse one small MLP for several cheap “extra outputs,” and fold legacy scalars / backout into learnable MUDD biases so we can squeeze **loss per step** without paying much wall-clock.

```bash
losses = [2.9161, 2.9165, 2.9176, 2.9179, 2.9172, 2.9171]
times = [860467, 860193, 860109, 860808, 860679, 860614]

print("p=%.4f" % scipy.stats.ttest_1samp(losses, 2.92, alternative="less").pvalue)
# p=0.0001

print("losses:", torch.std_mean(torch.tensor(losses)))
# losses: (tensor(0.0007), tensor(2.9171))

print("time:", torch.std_mean(torch.tensor(times)))
# time: (tensor(277.5980), tensor(860478.3125))

```

### CODE REVIEW

**Comment by shikhar-srivastava** [comment]
Hi @ClassicLarry -- are medium track WRs not being merged/maintained?
